### ISSUE TITLE
InstallData script created int/boolean customer attribute, visually does not reflects the on value in customer edit adminhtml

### ISSUE THREAD

**Issue by GianniPDF** [magento/magento2#25748]
### Preconditions (*)
1. Magento 2.3.3
2. PHP 7.2.24

### Steps to reproduce (*)
1. Create int/boolean customer attribute:
`Class InstallData implements InstallDataInterface{
const CUSTOMER_ATTR = 'customer_attr';
public function __construct(
\Magento\Eav\Setup\EavSetupFactory $eavSetupFactory,
\Magento\Eav\Model\Config $eavConfig
){
$this->eavSetupFactory = $eavSetupFactory;
$this->eavConfig = $eavConfig;
}

public function install(
\Magento\Framework\Setup\ModuleDataSetupInterface $setup,
\Magento\Framework\Setup\ModuleContextInterface $context
){
$eavSetup = $this->eavSetupFactory->create(['setup' => $setup]);
$eavSetup->addAttribute(
\Magento\Customer\Model\Customer::ENTITY,
self::CUSTOMER_ATTR,
[
'label' => 'Customerattr',
'type' => 'int',
'input' => 'boolean',
'source' => 'Magento\Eav\Model\Entity\Attribute\Source\Boolean',
'required' => false,
'system' => false,
'adminhtml_only' => 1,
'position' => 999
]
);
$attribute = $this->eavConfig->getAttribute(\Magento\Customer\Model\Customer::ENTITY, self::CUSTOMER_ATTR);
$attribute->addData([
'used_in_forms' => [
'adminhtml_customer'
]
]);
$attribute->save();
}
}`

2. Edit the input in customer adminhtml edit form and save
3. Check customer_entity_int table to see the value is saved
4. Check the customer adminhtml edit form to see that the input is not reflecting the on state

### Expected result (*)
1. Visually, the on value is reflected in the input

### Actual result (*)
1. Visually, the on value is not reflected in the input

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Echo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29294` were created

**Issue Available**: @engcom-Echo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by wagento-bhavin** [comment]
I've tested above script in 2.4-develop branch. It is working fine there. I also verified it in "customer_entity_int" table, the value is reflecting as per the option selected in admin customer edit form.

**Testing scenario:**

**Steps to reproduce:**

- Create new extension and put above attribute creation script in setup/InstallData.php
- Run setup upgrade command
- Edit any existing custom in admin panel and try to change the value of attribute
- Save the customer

**Result:**
No errors. Attribute value shows on/off on selecting and stores correct value in the database.
